### ISSUE TITLE
ner pipeline: consolidate ambiguity arising from analiticcl and extract entities

### ISSUE THREAD

**Issue by proycon** [knaw-huc/golden-agents-htr#7]
After running analiticcl, there may still be a fair amount of ambiguity in the data. Words (or phrases) are tagged with a class according to the lexicon they were found in, but this is often not clear cut and multiple classes may have been assigned, occasionally even with equal or very similar score.

There are however certain sequences of classes that are more likely, for example:

 * A first name followed by a family name -> stronger evidence for a full name
 * A material or property followed by an object -> stronger evidence for an object
 * A location after certain prepositions 
 * A first name followed by another first name -> second is most likely to be a family name instead?
 * Word at beginning of sentence as a first name but could very likely be an object if it were not for the initial casing?

I'd rather implement this outside of analitticcl in the ner pipeline (or a seperate tool invoked by the NER pipeline), so we don't increase the complexity of analiticcl itself. I'm still contemplating what the best approach is here. A machine learning approach would require training data, and I'm not sure if we can find enough that fits our use case nicely. Simple rule-based approaches are also conceivable.

**Comment by proycon** [comment]
I'm going for an approach inside analiticcl after all, because there I can do it most efficiently/compactly, the performance impact would be bigger otherwise. I'm aiming for a simple rule-based approach first where we simply specify combinations of tags to be favoured or penalised, this small bonus or penalty is then settled with the variant cost/score for the entire sequence.

**Comment by proycon** [comment]
My new implementation works in the initial unit/integration tests I wrote. How it holds in our actual golden agents scenario still remains to be seen. The new feature is documented [here](https://github.com/proycon/analiticcl#context-rules).

I'm considering expanding this implementation a bit and moving all the tagging functionality from the NER wrapper in this repo to analiticcl itself. As we've turned into that direction anyway in this issue, we better take it to its logical conclusion. The task of the NER wrapper will then remain to process the PageXML input, pass it to analiticcl, and convert its output to fully-fledged web annotations.
